pkg/controller/backup-operator/sync.go

```diff
@@ -66,25 +71,144 @@ func (b *Backup) processItem(key string) error {
 	}
 
 	eb := obj.(*api.EtcdBackup)
+
+	if eb.DeletionTimestamp != nil {
+		b.deletePeriodicBackupRunner(eb.ObjectMeta.UID)
+		err := b.removeFinalizerOfPeriodicBackup(eb)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
 	// don't process the CR if it has a status since
 	// having a status means that the backup is either made or failed.
-	if eb.Status.Succeeded || len(eb.Status.Reason) != 0 {
+	if !isPeriodicBackup(&eb.Spec) &&
+		(eb.Status.Succeeded || len(eb.Status.Reason) != 0) {
 		return nil
 	}
-	bs, err := b.handleBackup(&eb.Spec)
-	// Report backup status
-	b.reportBackupStatus(bs, err, eb)
+
+	if isPeriodicBackup(&eb.Spec) && b.isChanged(eb) {
+		// Stop previous backup runner if it exists
+		b.deletePeriodicBackupRunner(eb.ObjectMeta.UID)
```

Comment:
why check Spec with func isPeriodicBackup each time can we define var and check value?